File: M3Tools/Exceptions/Users/UsernameException.vb
﻿Namespace Exceptions
	<Serializable>
	Public Class UsernameException
        Inherits UserException

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(message As String)
            MyBase.New(message)
        End Sub

        Public Sub New(message As String, innerException As Exception)
            MyBase.New(message, innerException)
        End Sub
    End Class
End Namespace

File: M3Tools/Exceptions/Users/UsernameException.cs
using System;

namespace SPPBC.M3Tools.Exceptions
{
	/// <summary>
	/// An exception related to username operations
	/// </summary>
	[Serializable]
	public class UsernameException : UserException
	{

		/// <summary>
		/// 
		/// </summary>
		public UsernameException() : base()
		{
		}

		/// <summary>
		/// 
		/// </summary>
		/// <param name="message"></param>
		public UsernameException(string message) : base(message)
		{
		}

		/// <summary>
		/// 
		/// </summary>
		/// <param name="message"></param>
		/// <param name="innerException"></param>
		public UsernameException(string message, Exception innerException) : base(message, innerException)
		{
		}
	}
}
